### PR TITLE
Jmstall/yamlpoco

### PR DESCRIPTION
**Pull request by MikeStall**
Introduce a Schema object for strongly typed parameters. 
This defines the schema class in the ALM layer as the source of truth and lets document server include it.

### CODE REVIEW

File: src/PAModel/PAConvert/Yaml/YamlPocoSerializer.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            WriteCanonicalList(yaml, list);
+        }
+
+        // Write a list of properties (such as an object or dictionary) in an ordered way.
+        private static void WriteCanonicalList(YamlWriter yaml, List<KeyValuePair<string, object>> list)
+        {
+            // Critical to sort to preserve a canonical order.
+            list.Sort((kv1, kv2) => kv1.Key.CompareTo(kv2.Key));
+
+            foreach(var kv in list)
```
Comment (lesaltzm) [inline src/PAModel/PAConvert/Yaml/YamlPocoSerializer.cs:132]: nit: run autoformat on this file

File: src/PAModel/Schemas/Schema.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.PowerPlatform.Formulas.Tools.Schemas
+{
+    // Define Schema.yaml - strong typing for input and output parameters
+    public class ParameterSchema
+    {
+        // What names here?
```
Comment (lesaltzm) [inline src/PAModel/Schemas/Schema.cs:13]: nit: remove comment?

File: src/PAModel/Schemas/Schema.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+            // Optional: For entities, specifies the name
+            public string Name { get; set; }
+
+            // Optional: For records, can become recursive
+            public Dictionary<string, Parameter> Fields { get; set; }
+        }
+                
+        // Input parameters (public)
+        public Dictionary<string, Parameter> Parameters { get; set; }
```
Comment (lesaltzm) [inline src/PAModel/Schemas/Schema.cs:43]: Do we need all of Parameters/Outputs/Globals to begin with? 
This is an expansion direction, starting out I think we're only focusing on Parameters

**Review by lesaltzm — COMMENTED** [review]
Please add a more meaningful title and description as well

File: src/PAModel/Schemas/Schema.cs (hunk 3 above)
Comment (MikeStall) [reply src/PAModel/Schemas/Schema.cs:43]: It's a schema, so let's lay it out now.

**Review by lesaltzm — APPROVED** [review]

**Pull request merged**